src/main/java/it/gov/pagopa/pu/sil/service/AccessTokenService.java

```diff
@@ -0,0 +1,42 @@
+package it.gov.pagopa.pu.sil.service;
+
+import it.gov.pagopa.pu.auth.dto.generated.AccessToken;
+import it.gov.pagopa.pu.organization.dto.generated.OrgSilService;
+import it.gov.pagopa.pu.sil.service.legacyauth.SilLegacyAuthFacadeService;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@Service
+public class AccessTokenService {
+  private final SilLegacyAuthFacadeService silLegacyAuthFacadeService;
+
+  private final Map<Long, Pair<LocalDateTime, AccessToken>> clientId2accessTokensMap = new ConcurrentHashMap<>();
+
+  public AccessTokenService(SilLegacyAuthFacadeService silLegacyAuthFacadeService) {
+    this.silLegacyAuthFacadeService = silLegacyAuthFacadeService;
+  }
+
+  public AccessToken getAccessToken(OrgSilService orgSilService, AccessToken loggedUserAccessToken) {
+    if (orgSilService.getFlagLegacy().equals(Boolean.FALSE)) {
+      log.info("Using current access token {} for orgSilServiceId: {}", loggedUserAccessToken, orgSilService.getOrgSilServiceId());
```

Comment:
never print secret or personal data!
```suggestion
      log.debug("Using current access token for orgSilServiceId: {}", orgSilService.getOrgSilServiceId());
```

Reply:
I suggested a different message in order to avoid to print the token! it's a secret, it cannot be printed